agent/util-scripts/pbench-tool-meister

```diff
@@ -0,0 +1,836 @@
+    except Exception as exc:
+        logger.error(
+            "Unable to fetch and decode parameter key, '%s': %s", param_key, exc
+        )
+        return 4
+
+    pidfile_name = "{}.pid".format(PROG)
+    pfctx = pidfile.PIDFile(pidfile_name)
+    with daemon.DaemonContext(
+        stdout=sys.stdout,
+        stderr=sys.stderr,
+        working_directory=os.getcwd(),
+        umask=0o022,
+        pidfile=pfctx,
+        files_preserve=[sys.stdout, sys.stderr, fh.stream.fileno()],
+    ):
+        try:
+            # Previously we validated the tool meister parameters, and in
+            # doing so made sure we had proper access to the redis server.
+            #
+            # We can safely create the ToolMeister object now that we are
+            # "daemonized".
+            try:
+                # NOTE: we have to recreate the connection to the redis
+                # service since all open file descriptors were closed as part
+                # of the daemonizing process.
+                redis_server = redis.Redis(host=redis_host, port=redis_port, db=0)
+            except Exception as e:
+                logger.error(
+                    "Unable to connect to redis server, %s:%s: %s",
+                    redis_host,
+                    redis_port,
+                    e,
+                )
+                return 2
+            try:
+                tm = ToolMeister(pbench_bin, params, redis_server, logger)
+            except Exception:
+                logger.exception(
+                    "Unable to re-construct the ToolMeister" ' object with parms, "%r"',
+                    params,
+                )
+                return 4
+
+            terminate = False
+            try:
+                while not terminate:
+                    try:
+                        logger.debug("waiting ...")
+                        action, data = tm.wait_for_command()
+                        logger.debug("acting ... %r, %r", action, data)
+                        failures = action(data)
+                        if failures > 0:
+                            logger.warning(
+                                "%d failures encountered for action, '%r',"
+                                " on data, '%r'",
+                                failures,
+                                action,
+                                data,
+                            )
+                    except Terminate:
+                        logger.info("terminating")
+                        terminate = True
+            except Exception:
+                logger.exception("Unexpected error encountered")
+            finally:
+                tm.cleanup()
+        finally:
+            logger.info("Remove pid file ... (%s)", pidfile_name)
+            try:
+                os.unlink(pidfile_name)
+            except Exception:
+                logger.exception("Failed to remove pid file %s", pidfile_name)
+
+    return 0
+
+
```

Comment:
systemd units would be nice as well

Reply:
As mentioned before, using `systemd` would start to communicate a host vs benchmark instance governance conflict which would not be helpful.